commands/deploy.go

```diff
@@ -277,3 +300,87 @@ func compileEnvironment(envvarOpts []string, yamlEnvironment map[string]string,
 	functionAndStack := mergeMap(yamlEnvironment, fileEnvironment)
 	return mergeMap(functionAndStack, envvarArguments), nil
 }
+
+type authConfig struct {
+	Auth string `json:"auth,omitempty"`
+}
+
+type configFile struct {
+	AuthConfigs      map[string]authConfig `json:"auths"`
+	CredentialsStore string                `json:"credsStore,omitempty"`
+}
+
+const (
+	// docker default settings
+	configFileName        = "config.json"
+	configFileDir         = ".docker"
+	defaultDockerRegistry = "https://index.docker.io/v1/"
+)
+
+var (
+	configDir = os.Getenv("DOCKER_CONFIG")
+)
+
+func readDockerConfig(config *configFile) error {
```

Comment:
If this is copied from the Docker CLI it should probably be in a new package (i.e. sub folder and package)

Reply:
It was influenced by this: https://github.com/docker/cli/blob/master/cli/config/credentials/native_store.go, but it uses a different approach to read the creds and add them to the `config.AuthConfigs[n].Auth` for later use

Reply:
It uses the github.com/docker/docker-credential-helpers/credentials package